test/images/agnhost/netexec/netexec.go

```diff
@@ -125,7 +144,14 @@ func startHTTPServer(httpPort int) {
 	// older handlers
 	http.HandleFunc("/hostName", hostNameHandler)
 	http.HandleFunc("/shutdown", shutdownHandler)
-	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", httpPort), nil))
+
+	if err := server.ListenAndServe(); err != nil {
+		if err == http.ErrServerClosed {
```

Comment:
is the `server.Shutdown` call the only way we could get here? want to avoid blocking on the select{} if exitCh wasn't actually closed

Reply:
Yes.  There is no way to get ErrServerClosed without Shutdown.